File: verify_home.py
from playwright.sync_api import sync_playwright
import time

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page()
        try:
            print("Navigating to http://localhost:8081")
            page.goto("http://localhost:8081", timeout=60000)

            # Wait for key elements
            print("Waiting for content...")
            page.wait_for_selector("text=Tiendas en Liquidación", timeout=60000)

            # Take screenshot of top
            print("Taking screenshot 1...")
            page.screenshot(path="verification_home_top.png")

            # Scroll down to find the vertical list (Restaurants)
            # The vertical list is below the trending/rewards sections.
            # I'll scroll by 800 pixels.
            print("Scrolling down...")
            page.mouse.wheel(0, 800)
            time.sleep(2) # Wait for animation/render

            # Take screenshot of list
            print("Taking screenshot 2...")
            page.screenshot(path="verification_home_scroll.png")
            print("Screenshots saved.")

        except Exception as e:
            print(f"Error: {e}")
            page.screenshot(path="verification_error_scroll.png")
        finally:
            browser.close()

if __name__ == "__main__":
    run()

File: verify_home.ts
import { chromium } from 'playwright';

async function verify() {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Navigate to the local server
  try {
    await page.goto('http://localhost:8081', { waitUntil: 'networkidle' });

    // Wait for the main content to load
    await page.waitForSelector('text=Liquidación Total', { timeout: 10000 });

    // Take a screenshot of the top
    await page.screenshot({ path: 'verification_home.png' });
    console.log('Top screenshot captured.');

    // Scroll down to see the list of stores
    // "Últimas unidades" is the trending section. The list is below that.
    // We scroll enough to bypass the banner (approx 200px) and categories (approx 100px) and trending (approx 200px).
    await page.evaluate(() => window.scrollBy(0, 800));

    // Wait a bit for lazy loading / animation
    await page.waitForTimeout(2000);

    await page.screenshot({ path: 'verification_home_list.png' });
    console.log('List screenshot captured.');

  } catch (error) {
    console.error('Verification failed:', error);
  } finally {
    await browser.close();
  }
}

verify();
